src/app/upload/view.tsx

```diff
@@ -0,0 +1,210 @@
+"use client";
+
+import { Session } from "next-auth";
+import { signOut } from "next-auth/react";
+import { NextPage } from "next/types";
+import { useEffect, useState } from "react";
+import { Button } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import CloudUploadIcon from "@mui/icons-material/CloudUpload";
+import { uploadService } from "~/services/upload";
+import { api } from "~/trpc/react";
+import { redirect } from "next/navigation";
+import { Track } from "@prisma/client";
+import LogoutIcon from "@mui/icons-material/Logout";
+
+type Props = {
+  session: Session | null;
+};
+export const UploadView: NextPage<Props> = ({ session }) => {
+  const [musicUpload, setMusicUpload] = useState<File>();
+  const [progress, setProgress] = useState<{
+    percents: number;
+    operation: string;
+  }>({ operation: "", percents: 0 });
+  const [src, setSrc] = useState<string>("");
+
+  const [tracks, setTracks] = useState<Array<Track>>([]);
+  const uploadTracks = api.tracks.addFile.useMutation();
+  const getTracks = api.tracks.getFiles.useMutation();
+  const deleteTracks = api.tracks.deleteFile.useMutation();
+  const updateTracks = api.tracks.updateFile.useMutation();
+
+  useEffect(() => {
+    if (!session?.user || !session) redirect("/login");
```

Comment:
use `Next.js` middleware to protect pages
[Read more](https://next-auth.js.org/tutorials/securing-pages-and-api-routes#nextjs-middleware)